File: library/src/main/java/com/kimcy929/iconpakagereader/utils/RxSearch.java
package com.kimcy929.iconpakagereader.utils;

import android.text.TextUtils;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.SearchView;

import io.reactivex.subjects.BehaviorSubject;

public class RxSearch {
    public static BehaviorSubject<String> stringQuery(@NonNull SearchView searchView) {

        BehaviorSubject<String> behaviorSubject = BehaviorSubject.create();

        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {

            @Override
            public boolean onQueryTextSubmit(String query) {
                behaviorSubject.hasComplete();
                searchView.clearFocus();
                return true;
            }

            @Override
            public boolean onQueryTextChange(String newText) {
                if (TextUtils.isEmpty(newText)) {
                    behaviorSubject.onNext("");
                } else {
                    behaviorSubject.onNext(newText);
                }
                return true;
            }
        });

        return behaviorSubject;
    }
}

File: library/src/main/java/com/kimcy929/iconpakagereader/utils/RxSearch.kt
package com.kimcy929.iconpakagereader.utils

import android.text.TextUtils
import androidx.appcompat.widget.SearchView
import io.reactivex.subjects.BehaviorSubject

object RxSearch {
    fun stringQuery(searchView: SearchView): BehaviorSubject<String> {

        val behaviorSubject = BehaviorSubject.create<String>()

        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {

            override fun onQueryTextSubmit(query: String): Boolean {
                behaviorSubject.hasComplete()
                searchView.clearFocus()
                return true
            }

            override fun onQueryTextChange(newText: String): Boolean {
                if (TextUtils.isEmpty(newText)) {
                    behaviorSubject.onNext("")
                } else {
                    behaviorSubject.onNext(newText)
                }
                return true
            }
        })

        return behaviorSubject
    }
}
